flock_agent/daemon/flock_logs.py

```diff
@@ -0,0 +1,140 @@
+import os
+import time
+import json
+
+from .api_client import FlockApiClient
+
+
+class FlockLog:
+    def __init__(self, common, lib_dir):
+        self.c = common
+        self.filename = os.path.join(lib_dir, "flock.log")
+        self.c.log("FlockLog", "__init__", self.filename)
+
+        # If the log file doesn't exist, create an empty one
+        if not os.path.exists(self.filename):
+            open(self.filename, "a").close()
+            os.chmod(self.filename, 0o600)
+
+    def log(self, flock_log_type, twig_ids=None):
+        with open(self.filename, "a") as f:
+            f.write(
+                json.dumps(
+                    {
+                        "type": flock_log_type,
+                        "twig_ids": twig_ids,
+                        "timestamp": int(time.time()),
+                    }
+                )
+                + "\n"
+            )
+        os.chmod(self.filename, 0o600)
+
+    def submit_logs(self):
+        # Keep track of the biggest timestamp we see
+        biggest_timestamp = self.c.global_settings.get("last_flock_log_timestamp")
+
+        # What's the results file's modified timestamp, before we start the import
+        try:
+            mtime = os.path.getmtime(self.filename)
+
+            # Load the log file
+            with open(self.filename, "r") as f:
+                lines = f.readlines()
+                if len(lines) > 0:
+                    self.c.log("FlockLog", "submit_logs", "{} lines".format(len(lines)))
+
+                    # Start an API client
+                    api_client = FlockApiClient(self.c)
+                    try:
+                        api_client.ping()
+                    except:
+                        self.c.log(
+                            "FlockLog",
+                            "submit_logs",
+                            "API is not configured properly",
+                            always=True,
+                        )
+                        return
```

Comment:
Nope. When submitting osquery logs there's the same error, and this used to pop up as a notification, but not anymore now that the logic is in the daemon.

Also, I think the error message "API is not configured properly" isn't actually accurate, because this happens if, for example, the user is having a network problem or isn't connected to the internet, or if the server is temporarily down. I'm changing this message to "Unable to communicate with the server".